proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/HubResponseResource.java

```diff
@@ -19,72 +16,60 @@
-            @Session HttpSession session) {
+        @FormParam(SamlFormMessageType.SAML_RESPONSE) String hubResponse,
+        @FormParam("RelayState") String relayState,
+        @Session HttpSession session) {
 
-        try {
-            String expectedRequestId = (String) session.getAttribute("gateway_request_id");
-            hubResponseValidator.validate(encryptedHubResponse);
+        String hubRequestId = session.getAttribute(SESSION_KEY_HUB_REQUEST_ID).toString();
+        String eidasRequestId = session.getAttribute(SESSION_KEY_EIDAS_REQUEST_ID).toString();
+        String connectorEncrpytionCredential = session.getAttribute(SESSION_KEY_EIDAS_CONNECTOR_PUBLIC_CERT).toString();
 
-            if (!expectedRequestId.contains(encryptedHubResponse.getInResponseTo())) {
-                throw new InvalidHubResponseException(
-                    String.format("Received a Response from Hub for an AuthnRequest we have not seen (ID: %s)", encryptedHubResponse.getInResponseTo()));
-            }
+        LOG.info(String.format("[HUB Response] received for hub authn request ID '%s', eIDAS authn request ID '%s'", hubRequestId, eidasRequestId));
 
-            HubResponseContainer hubResponseContainer = HubResponseContainer.from(
-                    hubResponseValidator.getValidatedResponse(),
-                    hubResponseValidator.getValidatedAssertions()
-            );
+        HubResponseTranslatorRequest translatorRequest = new HubResponseTranslatorRequest(
+            hubResponse,
+            hubRequestId,
+            LEVEL_OF_ASSURANCE,
+            eidasRequestId,
+            connectorEncrpytionCredential
+        );
 
-            logHubResponse(hubResponseContainer);
-            Response eidasResponse = translatorService.getTranslatedResponse(encryptedHubResponse);
-            logEidasResponse(eidasResponse);
+        String eidasResponse = translatorProxy.getTranslatedResponse(translatorRequest);
 
-            return samlFormViewBuilder.buildResponse(
-                connectorNodeUrl,
-                eidasResponse,
-                "Post eIDAS Response SAML to Connector Node",
-                relayState
-            );
-        } catch (Throwable e) {
-            throw new HubResponseException(e, encryptedHubResponse);
-        }
-    }
+        LOG.info(String.format("[eIDAS Response] received for hub authn request ID '%s', eIDAS authn request ID '%s'", hubRequestId, eidasRequestId));
 
-    private void logHubResponse(HubResponseContainer hubResponseContainer) {
-        LOG.info("[Hub Response] ID: " + hubResponseContainer.getHubResponse().getResponseId());
-        LOG.info("[Hub Response] In response to: " + hubResponseContainer.getHubResponse().getInResponseTo());
-        LOG.info("[Hub Response] Provided level of assurance: " + hubResponseContainer.getAuthnAssertion().getProvidedLoa());
+        return samlFormViewBuilder.buildResponse(
+            connectorNodeUrl,
```

Comment:
Could we in future use the AuthnRequest `destination` instead of a static `connectorNodeUrl`?
This will have to be returned from `eidas-saml-parser`.

Reply:
I think the destination from the authn request would be the address of the gateway's authn request endpoint. Maybe we could get the ESP to return the issuer - I think that _might_ be the correct url?

Reply:
Actually I've had a look how the issuer is not the same as the endpoint the response should be sent to. The endpoints for current RP's transactions are defined in hub-federation-config, so I think we'll have to do a similar thing here and leave connectorNodeUrl in the config. The URL for the response isn't included in the authn request anywhere.

Reply:
Yep I think that's right. Looking at it this morning, we need the URL which is stored in the connector metadata. We'll extract that in the ESP

Reply:
James and i just confirmed the connector endpoint that the final response should be posted to can be pulled from the connector metadata. It'll be in the response DTO, currently as `destination`. We should probably change that though as it's not the same `destination` as in the authn request.